File: Java/src/main/java/org/codecop/lcdnumbers/DigitFactory.java
package org.codecop.lcdnumbers;

import java.util.Objects;

/**
 * Creates (LCD) digits from patterns.
 */
public class DigitFactory {

    private final Patterns patterns;

    public DigitFactory(Patterns patterns) {
        Objects.requireNonNull(patterns);

        this.patterns = patterns;
    }

    public ScalingDigit create(int digit) {
        return new ScalingDigit(digit, patterns);
    }
}

File: PHP/src/LcdNumbers/DigitFactory.php
<?php
namespace LcdNumbers;

/**
 * Creates (LCD) digits from patterns.
 */
class DigitFactory
{

    private $patterns;

    public function __construct(Patterns $patterns)
    {
        $this->patterns = $patterns;
    }

    public function create($digit)
    {
        return Digit::from($digit, $this->patterns);
    }
}
